apps/web/src/components/Dashboard/Templates.tsx

```diff
@@ -28,43 +59,128 @@ export function TemplatesContent({
+        `${process.env.NEXT_PUBLIC_API_URL}/templates?teamID=${teamId}`,
+        {
+          method: 'GET',
+          headers: {
+            Authorization: `Bearer ${user.accessToken}`,
+          },
+        }
+      )
+
+      const data: Template[] = await res.json()
+      if (!res.ok) {
+        toast({
+          title: 'An error occurred',
+          description: 'We were unable to fetch the team members',
         })
+        console.log(res.statusText)
+        return
       }
-    }
 
-    const interval = setInterval(() => {
-      f()
-    }, 5000)
+      // Display the most recently updated templates first
+      const orderedData = data.sort((a, b) => {
+        return new Date(b.updatedAt).getTime() - new Date(a.updatedAt).getTime()
+      })
+      setTemplates(orderedData)
+    }
 
-    f()
-    // Cleanup interval on component unmount
-    return () => clearInterval(interval)
+    fetchTemplates()
   }, [user, teamId])
 
+  async function deleteTemplate() {
+    const res = await fetch(
+      `${process.env.NEXT_PUBLIC_API_URL}/templates/${currentTemplate?.templateID}`,
+      {
+        method: 'DELETE',
+        headers: {
+          Authorization: `Bearer ${user.accessToken}`,
+        },
+      }
+    )
+
+    if (!res.ok) {
+      return toast({
+        title: 'An error occurred',
+        description: 'We were unable to delete the template',
+      })
+    }
+
+    setTemplates(
+      templates.filter(
+        (template) => template.templateID !== currentTemplate?.templateID
+      )
+    )
+    setCurrentTemplate(null)
+    setIsDeleteTemplateDialogOpen(false)
+  }
+
+  async function publishTemplate() {
+    const res = await fetch(
+      `${process.env.NEXT_PUBLIC_API_URL}/templates/${currentTemplate?.templateID}`,
+      {
+        method: 'PATCH',
+        headers: {
+          Authorization: `Bearer ${user.accessToken}`,
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({
+          public: !currentTemplate?.public,
+        }),
+      }
+    )
+
+    if (!res.ok) {
+      return toast({
+        title: 'An error occurred',
+        description: 'We were unable to update the template',
+      })
+    }
+
+    toast({
+      title: 'Template updated',
+      description: 'The template has been updated successfully',
+    })
+
+    setTemplates(
+      templates.map((template) =>
+        template.templateID === currentTemplate?.templateID
+          ? { ...template, public: !template.public }
+          : template
+      )
+    )
+    setCurrentTemplate(null)
+    setIsPublishTemplateDialogOpen(false)
+  }
+
   return (
     <div className="flex flex-col justify-center">
       <Table>
         <TableHeader>
-          <TableRow className="hover:bg-orange-500/10 dark:hover:bg-orange-500/10 border-b border-white/5 ">
+          <TableRow className="hover:bg-orange-500/10 dark:hover:bg-orange-500/10 border-b border-white/5">
+            <TableHead>Visibility</TableHead>
```

Comment:
https://e2b-team.slack.com/archives/C07UR0U4QD7/p1732745931793249?thread_ts=1732743673.013419&cid=C07UR0U4QD7

Reply:
I would probably rename to `Private` and do ✔️  or :x: